Automattic/wp-calypso
# Site-level billing management: Add /purchases/subscriptions/:site/:receiptId route

#### Changes proposed in this Pull Request

This implements the site-level view for a single subscription at the route ` /purchases/subscriptions/:site/:purchaseId`, mirroring the account-level route for a single subscription which exists at the route `/me/purchases/:site/:purchaseId`.

The new route will be available only with the feature flag `site-level-billing`, which is enabled for development and horizon only.

Related to (and builds upon) the site-level purchases list added in https://github.com/Automattic/wp-calypso/pull/45601.

Main project thread: pbOQVh-sc-p2


#### Testing instructions

- Visit `/purchases/subscriptions`.
- If you have multiple sites, verify that you see a site picker and that picking a site adds the site slug to the URL.
- Click on a purchase in the list.
- Verify that you are redirected to ` /purchases/subscriptions/:site/:purchaseId`, and that you see the purchase listed.
- Click the "Back" button in the header inside the page, and verify that you are redirected back to the site-level purchases list.
- Click the other links on the purchase page and verify that they all work as expected (TBD).

To view the new route directly, navigate to `/purchases/subscriptions/:site/:purchaseId`, where `:site` is the site URL and `:purchaseId` is the backend receipt ID (visible in payments admin).

Partially addresses #45679
